Add introduction and business use cases for workflows (#7729)
mendix/docs: content/en/docs/refguide/modeling/application-logic/workflows/_index.md

## 1 Introduction

Workflow is a visual language that allows you to build extendable processes. In Mendix Studio Pro, it is fully integrated with other visual editors, such as the microflow editor and page editor. 

> Implementing workflow processes is vital for organizations to enhance operational efficiency, ensure compliance with regulations, and mitigate risks. 

Reviewer: Very marketing style that we usually don't use in docs. 
```suggestion
Implementing workflow processes is beneficial for organizations that would like to enhance operational efficiency, ensure compliance with regulations, and mitigate risks. 
```

Below are the key reasons why organizations adopt workflow processes:

* Compliance management:
    * Ensuring adherence to local rules, regulations, and laws.

>     * Facilitating the documentation of any available data, including actions taken and decisions made. This helps to provide a clear audit trail.

Reviewer: >Facilitating the documentation of any available data
Is this grammatically correct? I don't understand what we are trying to say here, consider a rewrite?

Author: Rewritten in this commit: https://github.com/mendix/docs/pull/7729/commits/ad38649a7c8448c2f8b7125e42c0fb71504d6d24.

* Performance improvement:
    * Improving key performance indicators (KPIs) by identifying and addressing bottlenecks.

>     * Reducing the time to complete tasks, thereby minimizing overdue penalties.

Reviewer: ```suggestion
    * Reducing the time to complete tasks and minimizing overdue penalties.
```

>     * Enhancing the quality and efficiency of operations, leading to improved worker satisfaction.

Reviewer: ```suggestion
    * Enhancing the quality and efficiency of operations, which improves work satisfaction.
```

Reviewer: I can also suggest "which improves employee satisfaction".

* Risk reduction:
    * Preventing overdue penalties by streamlining processes.
    * Anticipating potential bottlenecks that could impact KPIs or introduce additional risks.
    * Avoiding overwork and stress among users and teams, thus maintaining compliance and productivity.

For more information on when and when not to use workflows, see the [When to Automate a Process with Workflows?](#when-to-use) and [When Not to Use Workflows?](#when-not) sections below.

For information on Mendix workflow use cases, see the [Mendix Workflow Use Cases](#use-case) section below.

> For instructions on how to perform Mendix workflow basic functions, see [Performing Workflow Basic Functions](/refguide/perform-workflow-basic-functions/).

Reviewer: I would mention this somewhere in the first paragraph.

Author: I moved it to the first paragraph and I also restructured the introduction a bit: https://github.com/mendix/docs/pull/7729/commits/ad38649a7c8448c2f8b7125e42c0fb71504d6d24.

> ## 2 When to Automate a Process with Workflows? {#when-to-use}

Reviewer: Is it worth aligning two sections and name this one "When to Use Workflows"? If you keep this title, I would remove the question mark.

Author: Yes, that's a good point! I've aligned the name and also removed the question mark. Thanks!

Automating a process with workflows is the most effective when the process meets specific criteria, ensuring smooth execution and optimization. 

Below are the criteria that a process should meet to be successfully implemented with workflows:

* Well-defined and known – the process is clearly understood and documented.
* Repeatable nature – the process involves tasks that are performed regularly or in a cyclical manner.
* High frequency of execution – the process occurs frequently, requiring consistent handling.
* Involvement of multiple participants – the process requires collaboration among different individuals or departments.
* Combination of human and automated tasks – the process involves a mix of activities that can be performed both by humans and automated systems.
* Stability – the process remains relatively unchanged over time, with minimal variations.
* Series of activities – it comprises a sequence of interconnected steps or actions.
* Longer duration – the process spans a significant period, ranging from hours to months.

> ## 3 When Not to Use Workflows? {#when-not}

Reviewer: ```suggestion
## 3 When Not to Use Workflows {#when-not}
```

While workflow processes offer numerous benefits, they may not always be the ideal solution for every scenario. 

Below are situations where the use of workflows might not be appropriate:

* Lack of applicability – if not MOST of the criteria outlined in the section above are met, it may not be a suitable use case for workflow implementation.
* High degree of collaboration and interaction – processes characterized by extensive collaboration and interaction among individuals, where no clear path can be defined, may not align well with workflow structures.
* Dynamic or unpredictable work – processes that involve dynamic or unknown elements, or where individuals require a high degree of freedom to decide their next actions, may not fit within the constraints of a workflow.
* Outcome-driven processes – processes focused more on achieving specific outcomes rather than following a predefined series of steps may not be best suited for workflow implementation. It might be beneficial to consider alternative approaches, such as utilizing multiple smaller workflows that are loosely coupled with other logic.

## 4 Mendix Workflow Use Cases {#use-case}

> Mendix offers a versatile development paradigm with multiple editors, enabling seamless incorporation of workflows into broader application design. With Mendix, you can keep workflows focused on representing your business processes, while using the other logic (such as microflows and nanoflows) for more technical implementations.

Reviewer: This style is very marketing, we don't usually use this style in docs.

Author: I've removed the marketing style in the same commit: https://github.com/mendix/docs/commit/ad38649a7c8448c2f8b7125e42c0fb71504d6d24

The table below presents a few use cases that are a good fit for Mendix workflows:

| Use case | Description |
| --- | --- |
| Supply chain management in regulated industries | Companies in highly regulated process industries can use Mendix workflows to manage supply chain changes, such as changes in packaging, weight, and other conditions. Mendix workflows validate and assess the impact of these changes at each stage, ensuring smooth adaptation to evolving requirements. |
| Migration from dedicated BPM platforms | Customers can transition from dedicated BPM platforms to Mendix to reduce cost, simplify maintenance, enhance user experience, and accelerate time-to-value. |

> | Maintenance plan approval | For customers who have migrated their industrial equipment maintenance plans from spreadsheets to Mendix, workflows facilitate to provide a structured review and approval process for changes of predefined maintenance plans. |

Reviewer: >workflows facilitate to provide
"Facilitate" is grammatically used with "something".

> | Maintenance plan approval | For customers who have migrated their industrial equipment maintenance plans from spreadsheets to Mendix, workflows facilitate to provide a structured review and approval process for changes of predefined maintenance plans. |

Reviewer: ```suggestion
| Maintenance plan approval | For customers who have migrated their industrial equipment maintenance plans from spreadsheets to Mendix, workflows provide a structured review and approval process for changes of predefined maintenance plans. |
```

> | Dynamic business process support for a new product launch | For companies needing rapid application development to support end-to-end business processes for new product launches, Mendix workflows can be seamlessly integrated for the subsequent legal and financial review and approval. |

Reviewer: ```suggestion
| Dynamic business process support for a new product launch | For companies needing rapid application development to support end-to-end business processes for new product launches, Mendix workflows can be integrated for the subsequent legal and financial review and approval. |
```

Reviewer: I am removing some of the marketing style.

> | Automated customer onboarding process | Organizations that need to streamline customer onboarding can use Mendix workflows to automate process orchestration across multiple systems and involve user intervention for invalid situations. This ensures a smooth and efficient onboarding experience. |

Reviewer: ```suggestion
| Automated customer onboarding process | Organizations that need to streamline customer onboarding can use Mendix workflows to automate the process across multiple systems and involve user intervention for invalid situations. This ensures a smooth onboarding experience. |
```

> To gain hands-on experience with Mendix workflow using a concrete use case, follow [Configuring a Workflow in Studio Pro for the Employee Onboarding Process](/refguide/workflow-how-to-configure/).

Reviewer: ```suggestion
For more information on a concrete use case, see [Configuring a Workflow in Studio Pro for the Employee Onboarding Process](/refguide/workflow-how-to-configure/).
```

## 1 Introduction

Workflow is a visual language that allows you to build extendable processes. It is fully integrated with other visual languages, such as the microflow editor and page editor. Implementing workflow processes is vital for organizations to enhance operational efficiency, ensure compliance with regulations, and mitigate risks. 

### 1.1 Why Do Organizations Implement Workflow Processes?

Below are the key reasons why organizations adopt workflow processes:

* Compliance management:
    * Ensuring adherence to local rules, regulations, and laws.

>     * Facilitating the documentation of actions taken, decisions made, and available data, providing a clear audit trail.

Reviewer 2: > decisions made, and available data

pls review this part.

Author: @PaulJongenMx I don't find this part problematic though. I still rephrased this part into the following:

> Facilitating the documentation of taken actions, decisions, and available data to provide a clear audit trail.

Please let me know if you have a different approach in mind.

Reviewer 2: For me as a reader it's unclear how to interpret the ", and" in  ", and available ...". And then after there is ", providing". Again a bit unclear how it related to the previous. With all the commas I have the impression I need to read this as a listing of things, but it's not. It's unclear how the parts between commas fit together.

Author: Understood!
Does the following sound better: 
Facilitating the documentation of any available data, including actions taken and decisions made. This helps to provide a clear audit trail.

* Performance improvement:

>     * Enhancing Key Performance Indicators (KPIs) by identifying and addressing bottlenecks.

Reviewer 2: "Improving KPI" is better I think

    * Reducing the time to complete tasks, thereby minimizing overdue penalties.
    * Enhancing the quality and efficiency of operations, leading to improved worker satisfaction.

* Risk reduction:
    * Preventing overdue penalties by streamlining processes.
    * Anticipating potential bottlenecks that could impact KPIs or introduce additional risks.
    * Avoiding overwork and stress among users and teams, thus maintaining compliance and productivity.

### 1.2 When to Automate a Process with Workflows?

Automating a process with workflow is most effective when the process meets specific criteria, ensuring smooth execution and optimization. By automating such processes, organizations can streamline operations, reduce errors, enhance efficiency, and free up human resources for more strategic tasks, ultimately driving productivity and achieving better outcomes. 

> 

Reviewer 2: Part if this is already mentioned above. Think it can be shorter.

> Here is when a process can be successfully automated:

Reviewer 2: when a process can be implemented using workflow.

Pls use the word workflow and not automated. Automated is too broad.

* Well-defined and known – the process is clearly understood and documented.
* Repeatable nature – the process involves tasks that are performed regularly or in a cyclical manner.
* High frequency of execution – the process occurs frequently, requiring consistent handling.
* Involvement of multiple participants – the process requires collaboration among different individuals or departments.
* Combination of human and automated tasks – it involves a mix of activities that can be performed both by humans and automated systems.
* Stability – the process remains relatively unchanged over time, with minimal variations.
* Series of activities – it comprises a sequence of interconnected steps or actions.

> * Longer duration – the process spans a significant period, ranging from hours to months, without significant alterations.

Reviewer 2: remove: "without significant alterations."

### 1.3 When Not to Use Workflows?

While workflow processes offer numerous benefits, they may not always be the ideal solution for every scenario. 

Here are situations where the use of workflows might not be appropriate:

> * Lack of applicability – if not all the criteria outlined in the section above for automating a process are met, it may not be a suitable use case for workflow implementation.

Reviewer 2: Should be: If not MOST of the  ....

* High degree of collaboration and interaction – processes characterized by extensive collaboration and interaction among individuals, where no clear path can be defined, may not align well with workflow structures.
* Dynamic or unpredictable work – processes that involve dynamic or unknown elements, or where individuals require a high degree of freedom to decide their next actions, may not fit within the constraints of a workflow.
* Outcome-driven processes – processes focused more on achieving specific outcomes rather than following a predefined series of steps may not be best suited for workflow implementation. It might be beneficial to consider alternative approaches, such as utilizing multiple smaller workflows that are loosely coupled with other logic.

### 1.4 Example Use Cases

> While many applications built on Mendix are data or transaction-centric, the integration of workflow capabilities varies based on specific requirements. Unlike dedicated BPM (Business Process Management) solutions where development revolves around process definitions, Mendix offers a versatile development paradigm with multiple editors, including the workflow editor, enabling seamless incorporation of workflows into broader application design. 

Reviewer 2: What are we trying to say here? The purpose of this section it to nicely separate us from the pure play workflow vendors where everything has to be a process.

Author: @PaulJongenMx I've revised this paragraph to the following:

> Mendix offers a versatile development paradigm with multiple editors, enabling seamless incorporation of workflows into broader application design. This is different from dedicated business process management (BPM) solutions where development has to revolve around process definitions.

What do you think?

Reviewer 2: THis is good, but would prefer to make this not a marketing text and call out to other products. I suggest we keep out the sentence "This is different from dedicated business process management (BPM) solutions where development has to revolve around process definitions."
Suggestion: 
"Mendix offers a versatile development paradigm with multiple editors, enabling seamless incorporation of workflows into broader application design. Not all applications need a workflow process to support a business process and often logic is technical by nature, i.e. API implementations. With mendix can you keep workflows focussed on the representation of your business process and use Microflow/nanoflow editor for more technical flow implementations."

Author: Thanks for the suggestion! 

Providing that this paragraph provides an introduction to Mendix workflow use cases (when to use workflows), I think it's better to remove this sentence "Not all applications need a workflow process to support a business process and often logic is technical by nature, i.e. API implementations.". 

I will use the following:

"Mendix offers a versatile development paradigm with multiple editors, enabling seamless incorporation of workflows into broader application design. With mendix, you can keep workflows focused on representing your business processes, while using the other logic (such as microflows and nanoflows) for more technical implementations."

Below presents a few example use cases that are a good fit for Mendix workflows:

* Maintenance plan approval:
    

>     A customer migrated their industrial equipment maintenance plans from spreadsheets to Mendix. While the core functionality revolves around these plans, instances requiring deviations necessitate a structured review and approval process facilitated by workflows.

Reviewer 2: "instances requiring deviations"  is a bit unclear. Can we mentioned "changes to predefined maintenance plans ...."

* Supply chain management in a regulated industry:
    
    A company in a highly regulated process industry leveraged Mendix to manage supply chain information changes like changes in packaging, weight and other conditions. Their application validates and assesses the impact of supply changes at each stage of the process, ensuring seamless adaptation to evolving requirements.

* Migration from dedicated BPM platforms:
    
    Customers transitioning from dedicated BPM platforms to Mendix for reasons such as cost reduction, simplified maintenance, enhanced user experience, and accelerated time-to-value.

* Dynamic business process support for a new product launch:
    
    A bank required a rapid application development to support an end-to-end business process for a new product launch. While the initial phase demanded dynamic collaboration between various teams, the subsequent legal and financial review and approval process was seamlessly integrated using Mendix workflows.

* Automated customer onboarding with orchestration:
    

>     An organization developed an application to streamline customer onboarding, involving orchestration across multiple systems. Although the process was highly automated, Mendix workflows were utilized to prompt user intervention in cases of invalid situations, ensuring a smooth onboarding experience.

Reviewer 2: Process orchestration was done using Mendix workflows and users were only intervention when the process identified anomalies.  This ensured a smooth and efficient onboarding experience.

Author: @PaulJongenMx I've rephrased this use case in the following way:

> Organizations that need to streamline customer onboarding can use Mendix workflows to automate process orchestration across multiple systems and prompt user intervention for invalid situations. This ensures a smooth and efficient onboarding experience.

What do you think?

Reviewer 2: Better. Not sure if this is common: "prompt user intervention"

Author: I will use "involve" instead of "prompt" then :)
